From aca959102d5e2a8bad93978ec62400f6b285ef72 Mon Sep 17 00:00:00 2001
From: TheGamerKing561 <eyadtamer2121@gmail.com>
Date: Mon, 23 Mar 2026 17:49:44 +0200
Subject: [PATCH] Use bms/capacity for battery percentage, keep battery/status
 for charging indicator

---
 twrp.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/twrp.cpp b/twrp.cpp
index c9f597a..d0e41e4 100644
--- a/twrp.cpp
+++ b/twrp.cpp
@@ -430,7 +430,7 @@ int main(int argc, char **argv) {
 			capacity_file += "/capacity";
 			FILE * cap = fopen(capacity_file.c_str(),"rt");
 #else
-			FILE * cap = fopen("/sys/class/power_supply/battery/capacity","rt");
+			FILE * cap = fopen("/sys/class/power_supply/bms/capacity","rt");
 #endif
 			if (cap) {
 				fgets(cap_s, 4, cap);
-- 
2.53.0

